### PR TITLE
Remove fs-extra dependency

### PR DESCRIPTION
**Pull request by lishaduck**
Manual rebase of #147.

Fixes: #120.
Supersedes/closes: #147.

### CODE REVIEW

File: lib/init.js [hunk 1]
```diff
@@ -6,7 +6,6 @@
 const path = require('node:path');
 const fs = require('graceful-fs');
 // TODO(@lishaduck) [engine:node@>=16.7.0]: Use native `cp`.
-const fsExtra = require('fs-extra');
```
Comment (jfmengels) [inline lib/init.js:9]: The comment above is now outdated, no?

File: lib/new-package.js [hunk 2]
```diff
@@ -455,11 +461,14 @@ function packageJson(options, packageName) {
       'update-examples': 'node maintenance/update-examples-from-preview.js',
       postinstall: 'elm-tooling install'
     },
-    dependencies: {
+    engines: {
+      node: '>=14.21.3'
+    },
+    devDependencies: {
```
Comment (jfmengels) [inline lib/new-package.js:467]: I'd keep these as dependencies. Elm packages that use `package.json` tend to have these as regular dependencies. Though I'm not sure that one is better than the other :shrug:

Comment (lishaduck) [reply lib/new-package.js:467]: I'd argue they're better as dependencies, but I'd meant to move these to another PR anyway.

**Comment by jfmengels** [comment]
One less dependency! Thank you :rocket:

**Pull request merged**